### PR TITLE
170 Possible filter UI improvement

### PR DESCRIPTION
**Pull request by luisabarca**
- Add a new toolbar for mobile devices
- Update component to show sorting options as a Select component
- Add button to open filter in a modal
- fix a z-index error for select component

### z-index bug
Screenshot 2023-02-01 at 23 44 12

### Video

https://user-images.githubusercontent.com/186492/216249206-e438abcd-94fd-4c9f-ae01-67faa0fc0e85.mov

### Before
nextjs-sanity formidable dev_products

### After
Screenshot 2023-02-02 at 0 27 03
Screenshot 2023-02-01 at 23 34 41

### CODE REVIEW

File: packages/nextjs/components/Button.tsx [hunk 1]
```diff
@@ -21,11 +23,11 @@ type ButtonAsLink = BaseProps &
 type ButtonProps = ButtonAsButton | ButtonAsLink;
 
 function ButtonComponent(
-  { as = "button", variant, disabled, className, ...props }: ButtonProps,
+  { as = "button", variant, disabled, className, leftIcon: LeftIcon, rightIcon: RightIcon, ...props }: ButtonProps,
```
Comment (sarmeyer) [inline packages/nextjs/components/Button.tsx:26]: I'm not sure we need this extra type param for `leftIcon` and `rightIcon`. Is there a reason you added it?

File: packages/nextjs/components/ProductSort.tsx [hunk 2]
```diff
@@ -2,8 +2,21 @@ import * as React from "react";
...
+type ProductSortProps = {
+  as?: "select" | "pills";
+  showTitle?: boolean;
+  title?: string;
+  selectClassName?: string;
+};
+
+export const ProductSort: React.FC<ProductSortProps> = ({
+  as = "pills",
+  showTitle = true,
```
Comment (sarmeyer) [inline packages/nextjs/components/ProductSort.tsx:16]: WDYT about making this `showTitle` prop `false` by default, it makes it a little clearer when seeing this component being used elsewhere when we have to explicitly add `showTitle` to the props

File: packages/nextjs/pages/products/index.tsx [hunk 3]
```diff
@@ -29,6 +29,9 @@ import { H6 } from "components/Typography/H6";
 import { Pagination } from "components/Pagination";
 import { FadeInOut } from "components/FadeInOut";
 import { Breadcrumbs } from "components/Breadcrumbs/Breadcrumbs";
+import { useDeviceSize } from "utils/useDeviceSize";
```
Comment (sarmeyer) [inline packages/nextjs/pages/products/index.tsx:32]: organizational nit: put this utils import up with the other ones

**Review by sarmeyer — APPROVED** [review]
This is great! I left a couple comments. Also, WDYT about adding a filter count to the filters button on smaller screen sizes? ex. `Filters (1)`

Just an idea!

File: packages/nextjs/components/Button.tsx (hunk 1 above)
Comment (luisabarca) [reply packages/nextjs/components/Button.tsx:26]: Yep, the left icon in the mobile toolbar

Screenshot 2023-02-02 at 16 26 54

File: packages/nextjs/components/ProductSort.tsx (hunk 2 above)
Comment (luisabarca) [reply packages/nextjs/components/ProductSort.tsx:16]: Yeah, sounds good, actually I was thinking to get rid of `showTitle` and just to validate `title.length`

Comment (luisabarca) [reply packages/nextjs/components/ProductSort.tsx]: Done

File: packages/nextjs/pages/products/index.tsx (hunk 3 above)
Comment (luisabarca) [reply packages/nextjs/pages/products/index.tsx]: Done

**Comment by luisabarca** [comment]
> This is great! I left a couple comments. Also, WDYT about adding a filter count to the filters button on smaller screen sizes? ex. Filters (1)

Sounds good, maybe we'll need another ticket, because we'll need to change the behavior of the checkboxes

File: packages/nextjs/components/Button.tsx (hunk 1 above)
Comment (sarmeyer) [reply packages/nextjs/components/Button.tsx:26]: Oh I mean having it renamed to `LeftIcon` and `RightIcon`.

Leaving the prop names as is works the same way, so I was wondering why you added the extra renaming

Comment (luisabarca) [reply packages/nextjs/components/Button.tsx:26]: Gotcha, yeah I was thinking to make it look like components prop names, but they are self-descriptive

Comment (luisabarca) [reply packages/nextjs/components/Button.tsx]: Thanks for the good catch ;), done!

**Pull request merged**